pkg/instancetype/apply/vmi.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+package apply
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	v1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	preferenceApply "kubevirt.io/kubevirt/pkg/instancetype/preference/apply"
+)
+
+type VMIApplier struct {
```

Comment:
Do I understand correctly, that this is a combined applier for instancetypes and preferences?

Reply:
Yeah this is just mimicing the existing behaviour of `InstancetypeMethods` at the moment but I'd like to break this down further in the future. Added to https://github.com/kubevirt/kubevirt/issues/13044.